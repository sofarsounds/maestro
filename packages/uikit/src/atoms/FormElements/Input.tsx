import React from 'react';
import styled, { css } from '../../lib/styledComponents';

import { inputBaseStyle, InputProps } from './Base';

import Icon from '../Icon';

interface SelectInputProps {
  isOpen?: boolean;
}

interface InputWrapper {
  children?: any;
  isOpen?: boolean;
}

interface Props {
  placeholder?: string;
  isOpen?: boolean;
}

interface ButtonProps {
  isOpen?: boolean;
  children?: any;
}

const Button = styled.button<ButtonProps>`
  flex-basis: 50px;
  ${inputBaseStyle};

  ${({ isOpen }) => isOpen && css``}
`;

const InputWrapper = styled.div<InputWrapper>`
  display: flex;
  flex: center;
`;

const InputStyle = styled.input<InputProps & SelectInputProps>`
  ${({ isOpen = false }) => css`
    ${inputBaseStyle};
    border-right: 0px;
    border-radius: 2px 0px 0px 2px;
    ${isOpen && css``}
  `}
`;

const Input: React.SFC<Props> = ({ isOpen, placeholder }) => (
  <InputWrapper>
    <InputStyle isOpen={isOpen} placeholder={placeholder} />
    <Button isOpen={isOpen}>
      <Icon name={isOpen ? 'caretDown' : 'check'} />
    </Button>
  </InputWrapper>
);

export default Input;
