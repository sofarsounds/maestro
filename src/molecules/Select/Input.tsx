import React from 'react';
import styled, { css } from '../../lib/styledComponents';

import { withTextfieldStyle, withShadow } from '../../util';
import { InputProps } from '../../typings/input';

import { SelectState } from './index';
import Icon from '../../atoms/Icon';
import Spinner from '../../atoms/Spinner';

interface SelectInputProps {
  isOpen?: boolean;
  readonly?: boolean;
  invertColor?: boolean;
}

interface Wrapper {
  children?: React.ReactNode;
  isOpen?: boolean;
  hasError?: boolean;
  invertColor?: boolean;
  readOnly?: boolean;
  initialWidth?: string;
}

interface Props {
  invertColor?: boolean;
  isOpen?: boolean;
  onToggle: () => void;
  innerRef?: React.RefObject<HTMLInputElement>;
  hasError?: boolean;
  renderLeftIcon?: React.ReactNode;
  state: SelectState;
  initialWidth?: string;
  'data-qaid'?: string;

  inputProps: {
    id?: string;
    placeholder?: string;
    readOnly?: boolean;
    name?: string;
    value?: string;
    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
    onClear?: () => void;
  };
}

interface ButtonProps {
  isOpen?: boolean;
  children?: React.ReactNode;
  invertColor?: boolean;
}

const Wrapper = styled.div<Wrapper>`
  ${({ theme, readOnly, invertColor, isOpen, initialWidth }) => css`
    display: flex;
    flex: center;
    align-items: center;
    ${withTextfieldStyle};
    padding-right: ${theme.ruler[2]}px;

    ${readOnly &&
      css`
        cursor: pointer;
      `}

    ${isOpen &&
      css`
        ${withShadow({ depth: 3 })}

        &:hover {
          border-color: ${theme.colors.macyGrey};
        }
      `}

    ${invertColor &&
      css`
        background: transparent;
        border-color: ${theme.colors.whiteDenim};

        &:hover {
          border-color: ${theme.colors.whiteDenim};
        }
        &:focus {
          border-color: ${theme.colors.silverSprings};
          background: rgba(219, 219, 219, 0.1); //macyGrey at 10%
        }
      `}

    ${initialWidth &&
      css`
        width: 100%;
        max-width: ${initialWidth};

        &:focus,
        &:hover,
        &:active {
          max-width: 100%;
        }

        ${isOpen &&
          css`
            max-width: 100%;
          `}
      `}
  `}
`;

const IconWrapper = styled.div<{ invertColor?: boolean }>`
  ${({ theme, invertColor }) => css`
    margin-right: ${theme.ruler[2]}px;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;

    i::before {
      color: ${theme.colors[invertColor ? 'whiteDenim' : 'blueSmoke']};
    }
  `}
`;

const StyledInput = styled.input<InputProps & SelectInputProps>`
  ${({ theme, invertColor, readOnly }) => css`
    background: transparent;
    height: 100%;
    width: 100%;
    border: none;
    font-size: ${theme.fontSizes.body2};
    letter-spacing: 0.1px;
    color: ${theme.colors.backToBlack};

    ${readOnly &&
      css`
        cursor: pointer;
      `}

    &:hover, &:focus {
      outline: none;
    }

    &::placeholder {
      color: ${theme.colors.blueSmoke};
    }

    ${invertColor &&
      css`
        color: ${theme.colors.whiteDenim};

        &::placeholder {
          color: ${theme.colors.whiteDenim};
        }
      `}
  `}
`;

const ActionButton = styled.button<ButtonProps>`
  ${({ theme, invertColor }) => css`
    width: 25px;
    height: 25px;
    border: none;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 0.15s ease;
    cursor: pointer;
    border-radius: 100%;

    &:hover {
      background-color: ${theme.colors[
        invertColor ? 'paintItBlack' : 'silverSprings'
      ]};
    }

    &:focus {
      outline: none;
    }

    i::before {
      color: ${theme.colors[invertColor ? 'whiteDenim' : 'blueSmoke']};
    }
  `}
`;

const Input: React.SFC<Props> = ({
  isOpen,
  onToggle,
  inputProps = {},
  innerRef,
  hasError,
  invertColor,
  renderLeftIcon,
  state,
  initialWidth,
  'data-qaid': qaId
}) => {
  const enableClearButton =
    state === SelectState.ready &&
    !inputProps.readOnly &&
    inputProps.onClear &&
    inputProps.value;

  return (
    <Wrapper
      id={inputProps.id}
      readOnly={inputProps.readOnly}
      tabIndex={0}
      hasError={hasError}
      onClick={onToggle}
      isOpen={isOpen}
      ref={innerRef}
      invertColor={invertColor}
      initialWidth={initialWidth}
      data-qaid={qaId}
    >
      {renderLeftIcon && (
        <IconWrapper invertColor={invertColor} data-qaid={`${qaId}-left-icon`}>
          {renderLeftIcon}
        </IconWrapper>
      )}
      <StyledInput
        // We need the selectable element to also be the trigger for the Portal - what's the best way to do this?
        tabIndex={-1}
        aria-haspopup="listbox"
        value={inputProps.value}
        readOnly={inputProps.readOnly}
        isOpen={isOpen}
        placeholder={inputProps.placeholder || 'Please Select'}
        name={inputProps.name}
        onChange={inputProps.onChange}
        invertColor={invertColor}
        data-qaid={`${qaId}-input`}
        autoComplete="nope"
      />

      {state === SelectState.loading && (
        <div style={{ width: '30px' }}>
          <Spinner data-qaid={`${qaId}-spinner`} size="20px" />
        </div>
      )}

      {state === SelectState.error && (
        <Icon
          data-qaid={`${qaId}-error-icon`}
          color="redRedWine"
          name="alertTriangle"
          title="Could not load options..."
        />
      )}

      {enableClearButton && (
        <ActionButton
          invertColor={invertColor}
          type="button"
          data-qaid={`${qaId}-clear`}
          onClick={inputProps.onClear}
          title="Clear"
        >
          <Icon data-qaid={`${qaId}-clear-icon`} name="close" />
        </ActionButton>
      )}

      <ActionButton
        invertColor={invertColor}
        isOpen={isOpen}
        type="button"
        data-qaid={`${qaId}-toggle`}
        title="Toggle"
      >
        <Icon
          data-qaid={`${qaId}-toggle-icon`}
          name={isOpen ? 'caretUp' : 'caretDown'}
        />
      </ActionButton>
      <div id="selectOptions" tabIndex={-1} role="listbox"></div>
    </Wrapper>
  );
};
export default Input;
