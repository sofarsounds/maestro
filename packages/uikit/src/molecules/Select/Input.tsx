import React from 'react';
import styled, { css } from '../../lib/styledComponents';

import { withShadow } from '../../util';
import { inputBaseStyle, InputProps } from '../../atoms/FormElements/Base';

import Icon from '../../atoms/Icon';

interface SelectInputProps {
  isOpen?: boolean;
  depth: any;
}

interface InputWrapper {
  children?: any;
  isOpen?: boolean;
}

interface Props {
  placeholder?: string;
  isOpen?: boolean;
  onClick: () => void;
  innerRef?: React.RefObject<any>;
  value?: any;
}

interface ButtonProps {
  isOpen?: boolean;
  children?: any;
}

const Button = styled.button<ButtonProps>`
  ${inputBaseStyle};
  flex-basis: 50px;
  border-radius: 0px 2px 2px 0px;

  ${({ isOpen }) => isOpen && css``}
`;

const InputWrapper = styled.div<InputWrapper>`
  display: flex;
  flex: center;
`;

const InputStyle = styled.input<InputProps & SelectInputProps>`
  ${({ theme, isOpen = false }) => css`
    ${inputBaseStyle};
    border-right: 0px;

    &::placeholder {
      color: ${theme.colours.blueSmoke};
    }

    border-radius: 2px 0px 0px 2px;
    ${isOpen &&
      css`
        ${withShadow}
      `}
  `}
`;

const Input: React.SFC<Props> = ({
  isOpen,
  placeholder,
  onClick,
  innerRef,
  value
}) => {
  return (
    <InputWrapper ref={innerRef}>
      <InputStyle
        value={value}
        depth={3}
        isOpen={isOpen}
        placeholder={placeholder}
      />
      <Button onClick={e => onClick()} isOpen={isOpen}>
        <Icon name={isOpen ? 'caretDown' : 'check'} />
      </Button>
    </InputWrapper>
  );
};

export default Input;
