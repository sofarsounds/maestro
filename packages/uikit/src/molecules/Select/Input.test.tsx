import React from 'react';
/* import theme from '../../theme'; */
import { mountWithTheme } from '../../test';
import Input from './Input';

const mockClick = jest.fn();

const setup = (
  isOpen: boolean = false,
  placeholder: string = 'I am a placeholder'
) =>
  mountWithTheme(
    <Input isOpen={isOpen} placeholder={placeholder} toggleSelect={mockClick}>
      I am a child component
    </Input>
  );

describe('Select <Input />', () => {
  it('it renders correctly', () => {
    expect(setup()).toMatchSnapshot();
  });

  it('it has the correct style properties when isOpen is false', () => {
    let wrapper = setup(false);
    let input = wrapper;
    expect(input).toHaveStyleRule('border-radius', '2px');
    expect(input).not.toHaveStyleRule('box-shadow');
  });

  /* TODO: Why this doesn't work */
  /* wrapper.setProps({ children: <Input isOpen={true} onClick={mockClick} /> }); */
  /* wrapper.update(); */
  /* input.update(); */

  it('it has the correct style properties when isOpen is true', () => {
    let wrapper = setup(true);
    let input = wrapper;
    expect(input).toHaveStyleRule('border-radius', '2px');
    expect(input).not.toHaveStyleRule('box-shadow');

    expect(input).toHaveStyleRule(
      'box-shadow',
      '0 10px 20px 0 rgba(0,0,0,0.19)'
    );
  });

  it('it renders a placeholder', () => {
    let wrapper = setup();
    let placeholder = wrapper.find('input').props().placeholder;
    expect(placeholder).toBe('I am a placeholder');
  });

  it('it triggers onClick when clicked', () => {
    let wrapper = setup().find('button');
    wrapper.simulate('click');
    expect(mockClick).toHaveBeenCalled();
  });

  it('it shows the up icon when isOpen is true', () => {
    let wrapper = setup(true);
    let icon = wrapper.find('Icon').props().name;
    expect(icon).toBe('caretUp');
  });

  it('it shows the down icon when isOpen is false', () => {
    let wrapper = setup(false);
    let icon = wrapper.find('Icon').props().name;
    expect(icon).toBe('caretDown');
  });
});
