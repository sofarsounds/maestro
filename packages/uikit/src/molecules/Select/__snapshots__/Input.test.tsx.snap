// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select <Input /> it renders correctly 1`] = `
.c3 {
  font-family: 'sofarsounds-icon-font' !important;
  speak: none;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  line-height: 1;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c3:before {
  color: #000;
}

.c3.icon-facebook:before {
  content: '\\e90d';
}

.c3.icon-instagram:before {
  content: '\\e916';
}

.c3.icon-spotify:before {
  content: '\\e926';
}

.c3.icon-twitter:before {
  content: '\\e92f';
}

.c3.icon-youtube:before {
  content: '\\e935';
}

.c3.icon-facebookOutline:before {
  content: '\\e90b';
}

.c3.icon-instagramOutline:before {
  content: '\\e911';
}

.c3.icon-spotifyOutline:before {
  content: '\\e914';
}

.c3.icon-twitterOutline:before {
  content: '\\e91a';
}

.c3.icon-youtubeOutline:before {
  content: '\\e91b';
}

.c3.icon-caretUp:before {
  content: '\\e921';
}

.c3.icon-caretDown:before {
  content: '\\e906';
}

.c3.icon-caretLeft:before {
  content: '\\e922';
}

.c3.icon-caretRight:before {
  content: '\\e923';
}

.c3.icon-caretCircleDown:before {
  content: '\\e908';
}

.c3.icon-chevronLeft:before {
  content: '\\e917';
}

.c3.icon-chevronRight:before {
  content: '\\e920';
}

.c3.icon-moveVertical:before {
  content: '\\e919';
}

.c3.icon-check:before {
  content: '\\e927';
}

.c3.icon-checkCircle:before {
  content: '\\e902';
}

.c3.icon-alertCircle:before {
  content: '\\e907';
}

.c3.icon-alertTriangle:before {
  content: '\\e930';
}

.c3.icon-calendar:before {
  content: '\\e901';
}

.c3.icon-time:before {
  content: '\\e929';
}

.c3.icon-timeHand:before {
  content: '\\e928';
}

.c3.icon-byob:before {
  content: '\\e900';
}

.c3.icon-close:before {
  content: '\\e903';
}

.c3.icon-commercial:before {
  content: '\\e904';
}

.c3.icon-disabled:before {
  content: '\\e905';
}

.c3.icon-eye:before {
  content: '\\e909';
}

.c3.icon-video:before {
  content: '\\e90e';
}

.c3.icon-happyMask:before {
  content: '\\e90f';
}

.c3.icon-helpCircleOutline:before {
  content: '\\e912';
}

.c3.icon-ellipsisHorizontal:before {
  content: '\\e918';
}

.c3.icon-no:before {
  content: '\\e91c';
}

.c3.icon-pet:before {
  content: '\\e91d';
}

.c3.icon-readGlasses:before {
  content: '\\e91e';
}

.c3.icon-residential:before {
  content: '\\e91f';
}

.c3.icon-tree:before {
  content: '\\e92b';
}

.c3.icon-heartPulse:before {
  content: '\\e90c';
}

.c3.icon-homeWorld:before {
  content: '\\e910';
}

.c3.icon-travelWorld:before {
  content: '\\e915';
}

.c3.icon-musicTicket:before {
  content: '\\e913';
}

.c3.icon-musicLove:before {
  content: '\\e90a';
}

.c2 {
  -webkit-flex-basis: 50px;
  -ms-flex-preferred-size: 50px;
  flex-basis: 50px;
  border: none;
}

.c2:hover {
  cursor: pointer;
}

.c2:focus {
  border-color: #DBDBDB;
  outline: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: center;
  -ms-flex: center;
  flex: center;
  width: 100%;
  height: 44px;
  background-color: #FFFFFF;
  border-radius: 2px;
  border: 1px solid #DBDBDB;
  padding: 0 16px;
  outline: none;
  box-shadow: none;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  font-size: 14px;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
  color: #111111;
  padding-right: 0px;
}

.c0:hover {
  border-color: #949494;
}

.c0:focus {
  border-color: #111111;
}

.c0:disabled {
  color: #DBDBDB;
  border-color: #DBDBDB;
}

.c1 {
  border-right: 0px;
  width: 100%;
  border: 0px;
  font-size: 14px;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
  color: #111111;
}

.c1:hover {
  outline: none;
}

.c1:focus {
  outline: none;
}

.c1::-webkit-input-placeholder {
  color: #949494;
}

.c1::-moz-placeholder {
  color: #949494;
}

.c1:-ms-input-placeholder {
  color: #949494;
}

.c1::placeholder {
  color: #949494;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "default": "2px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green800": "#008736",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#039842",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
        "navbarHeight": Object {
          "lg": "84px",
          "md": "74px",
          "xs": "62px",
        },
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "formGroupLabel": "14px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "54px",
      },
      "fonts": Object {
        "regular": "Open Sans",
      },
      "media": Object {
        "lg": [Function],
        "md": [Function],
        "sm": [Function],
        "xl": [Function],
        "xs": [Function],
      },
      "ruler": Array [
        0,
        4,
        8,
        12,
        16,
        20,
        24,
        28,
        32,
        36,
        40,
        44,
        48,
        52,
        56,
        60,
        64,
        68,
        72,
        76,
        80,
      ],
      "utils": Object {
        "transition": [Function],
      },
      "zIndex": Object {
        "navbar": 100,
      },
    }
  }
>
  <Input
    isOpen={false}
    placeholder="I am a placeholder"
    toggleSelect={[MockFunction]}
  >
    <styled.div
      depth={3}
      isOpen={false}
      onClick={[Function]}
    >
      <StyledComponent
        depth={3}
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "
  ",
                [Function],
                "
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        isOpen={false}
        onClick={[Function]}
      >
        <div
          className="c0"
          onClick={[Function]}
        >
          <styled.input
            isOpen={false}
            placeholder="I am a placeholder"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-ifAKCX",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  ",
                      [Function],
                      "
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-ifAKCX",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              isOpen={false}
              placeholder="I am a placeholder"
            >
              <input
                className="c1"
                placeholder="I am a placeholder"
              />
            </StyledComponent>
          </styled.input>
          <styled.button
            isOpen={false}
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-htpNat",
                    "isStatic": false,
                    "lastClassName": "c2",
                    "rules": Array [
                      "
",
                      [Function],
                      "
  }
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-htpNat",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              isOpen={false}
            >
              <button
                className="c2"
              >
                <Icon
                  name="caretDown"
                >
                  <styled.i
                    className="icon-caretDown"
                  >
                    <StyledComponent
                      className="icon-caretDown"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bwzfXH",
                            "isStatic": false,
                            "lastClassName": "c3",
                            "rules": Array [
                              "
  font-family: 'sofarsounds-icon-font' !important;
  speak: none;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  line-height: 1;

  /* Better Font Rendering =========== */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  &:before {
    color: ",
                              [Function],
                              ";
  }

  ",
                              [Function],
                              "

  ",
                              "
    ",
                              "
      &.icon-facebook:before {
        content: '\\\\e90d';
      }
    
      &.icon-instagram:before {
        content: '\\\\e916';
      }
    
      &.icon-spotify:before {
        content: '\\\\e926';
      }
    
      &.icon-twitter:before {
        content: '\\\\e92f';
      }
    
      &.icon-youtube:before {
        content: '\\\\e935';
      }
    
      &.icon-facebookOutline:before {
        content: '\\\\e90b';
      }
    
      &.icon-instagramOutline:before {
        content: '\\\\e911';
      }
    
      &.icon-spotifyOutline:before {
        content: '\\\\e914';
      }
    
      &.icon-twitterOutline:before {
        content: '\\\\e91a';
      }
    
      &.icon-youtubeOutline:before {
        content: '\\\\e91b';
      }
    
      &.icon-caretUp:before {
        content: '\\\\e921';
      }
    
      &.icon-caretDown:before {
        content: '\\\\e906';
      }
    
      &.icon-caretLeft:before {
        content: '\\\\e922';
      }
    
      &.icon-caretRight:before {
        content: '\\\\e923';
      }
    
      &.icon-caretCircleDown:before {
        content: '\\\\e908';
      }
    
      &.icon-chevronLeft:before {
        content: '\\\\e917';
      }
    
      &.icon-chevronRight:before {
        content: '\\\\e920';
      }
    
      &.icon-moveVertical:before {
        content: '\\\\e919';
      }
    
      &.icon-check:before {
        content: '\\\\e927';
      }
    
      &.icon-checkCircle:before {
        content: '\\\\e902';
      }
    
      &.icon-alertCircle:before {
        content: '\\\\e907';
      }
    
      &.icon-alertTriangle:before {
        content: '\\\\e930';
      }
    
      &.icon-calendar:before {
        content: '\\\\e901';
      }
    
      &.icon-time:before {
        content: '\\\\e929';
      }
    
      &.icon-timeHand:before {
        content: '\\\\e928';
      }
    
      &.icon-byob:before {
        content: '\\\\e900';
      }
    
      &.icon-close:before {
        content: '\\\\e903';
      }
    
      &.icon-commercial:before {
        content: '\\\\e904';
      }
    
      &.icon-disabled:before {
        content: '\\\\e905';
      }
    
      &.icon-eye:before {
        content: '\\\\e909';
      }
    
      &.icon-video:before {
        content: '\\\\e90e';
      }
    
      &.icon-happyMask:before {
        content: '\\\\e90f';
      }
    
      &.icon-helpCircleOutline:before {
        content: '\\\\e912';
      }
    
      &.icon-ellipsisHorizontal:before {
        content: '\\\\e918';
      }
    
      &.icon-no:before {
        content: '\\\\e91c';
      }
    
      &.icon-pet:before {
        content: '\\\\e91d';
      }
    
      &.icon-readGlasses:before {
        content: '\\\\e91e';
      }
    
      &.icon-residential:before {
        content: '\\\\e91f';
      }
    
      &.icon-tree:before {
        content: '\\\\e92b';
      }
    
      &.icon-heartPulse:before {
        content: '\\\\e90c';
      }
    
      &.icon-homeWorld:before {
        content: '\\\\e910';
      }
    
      &.icon-travelWorld:before {
        content: '\\\\e915';
      }
    
      &.icon-musicTicket:before {
        content: '\\\\e913';
      }
    
      &.icon-musicLove:before {
        content: '\\\\e90a';
      }
    ",
                              "
  ",
                              ";
",
                            ],
                          },
                          "displayName": "styled.i",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bwzfXH",
                          "target": "i",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <i
                        className="icon-caretDown c3"
                      />
                    </StyledComponent>
                  </styled.i>
                </Icon>
              </button>
            </StyledComponent>
          </styled.button>
        </div>
      </StyledComponent>
    </styled.div>
  </Input>
</ThemeProvider>
`;
